services/frontend/src/components/settings/settings-page.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import _ from 'lodash';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { TextField, Button } from '@material-ui/core';
+import { Page } from '../shared';
+import { setLoginStage } from '../../store/login';
+import UsernameModal from './username-modal';
+import DeleteModal from './delete-modal';
+import styles from './settings-page.module.scss';
+
+export class SettingsPage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.openNewUsernameModal = this.openNewUsernameModal.bind(this);
+        this.closeNewUsernameModal = this.closeNewUsernameModal.bind(this);
+        this.openNewDeleteModal = this.openNewDeleteModal.bind(this);
+        this.closeNewDeleteModal = this.closeNewDeleteModal.bind(this);
+
+        this.state = {
+            newUsernameModal: false,
+            newDeleteModal: false,
+        };
+    }
+
+    openNewUsernameModal() {
+        this.setState({ newUsernameModal: true });
+    }
+
+    closeNewUsernameModal() {
+        this.setState({ newUsernameModal: false });
+    }
+
+    openNewDeleteModal() {
+        this.setState({ newDeleteModal: true });
+    }
+
+    closeNewDeleteModal() {
+        this.setState({ newDeleteModal: false });
+    }
+
+    render() {
+        return (
+            <Page title="Settings">
+                <div className={styles.username}>
+                    <TextField
+                        label="Username"
+                        InputProps={{
+                            readOnly: true,
+                        }}
+                        value={_.get(this.props, 'user.username')}
+                    />
```

Comment:
You can just use [`Typography`](https://material-ui.com/components/typography/) to display this, since we're not actually using the input functionality.